WordPress/src/main/java/org/wordpress/android/ui/AppLogViewerActivity.java

```diff
@@ -136,15 +139,16 @@ private void copyAppLogToClipboard() {
     }
 
     @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
+    public boolean onCreateOptionsMenu(@NonNull Menu menu) {
         super.onCreateOptionsMenu(menu);
         MenuInflater inflater = getMenuInflater();
         inflater.inflate(R.menu.app_log_viewer_menu, menu);
         return true;
     }
 
     @Override
-    public boolean onOptionsItemSelected(final MenuItem item) {
```

Comment:
The only reason I removed this `final` keyword is because I wanted every such method across the codebase to be consistent with each other.

Thus, and because I found very few occurrences of such `final` usage on such `onOptionsItemSelected(...)` methods across the whole codebase, that it is better to remove it here as well, rather than adding it everywhere else. No matter, I think it is better for us to use a consistent signature for such methods. Wdyt? 🤔

PS: Also, and since we don't have a documented way of using `final`, and since I rarely seen it being used, I think it is better to avoid adding it unnecessarily everywhere. Thus, I decided it might be better to remove it instead, whenever found on those `@Override` methods.

Reply:
> I think it is better for us to use a consistent signature for such methods

I agree with this, but my point above is that `final` is not really part of the method signature. That's what I meant by "no effect at the call site".

Since, in this instance, the usage is for such a simple method, I think it's ok to remove it. Generally speaking, though, the _intention_ for using this is entirely compatible with keeping signatures consistent. `final` on a parameter only pertains to the usage of that reference _within the method body_, and is sometimes used to prevent reassignment of the variable. This can be a useful restriction in long / complex methods.

On this same topic, there are many parameter renaming changes, in which the commit message mentions this was done for the sake of consistent method signatures, but parameter names are not part of a methods signature in Java. I didn't see any issue with the renamings that you did, but was curious to know your motivation. Does it relate somehow to Kotlin?

To play devils advocate, I could envision deviating from the overriden super class parameter names for clarity's sake - to I don't think we should have a rule that all parameter names _must_ match that of the super class. Especially because some super methods have inappropriately terse parameter names.

Reply:
> I agree with this, but my point above is that final is not really part of the method signature. That's what I meant by "no effect at the call site".

👍 

> Generally speaking, though, the intention for using this is entirely compatible with keeping signatures consistent.

I agree with all your points @mkevins  and apologies for the confusion here. What I meant with keeping signatures consistent wan't related to super vs implementing signature, but rather project wise, that is, so that when one is searching/copy-pasting such methods, that they are not seeing a mixture of both, some of them using a `final`, while others are not. Now, I agree that the usage of `final` is a case-by-case scenario, but on those specific cases I doubt that it was done like that (an assumption from my side). Instead, I think this was initial done purely because it is a best practice to use `final` as much and as widely as possible, and then, that was blindly copy-pasted thereafter. However, since us using `final` vs not using it at all hasn't been discussed, or agreed upon, and since me seeing and experiencing us not using `final` during our day-to-day work with Java, that is, unless the compiler complains or warns us about it, I was thinking it is better to remove those usages as well. This way, IMHO, the next person to copy-paste such method would not unnecessarily copy-paste the `final` as well, even if that means safer code... 🤷

The other way around for me would have been to add the `final` to every such `@Override` method. 🤔

Having said all the above, it is easier for me to plain ignore such changes if this will make it easier for you to review. I am only doing those to increase the consistency and make sure that we are using such `@Override` method consistently across the codebase, just to have make us that bit more consistent. But, I can stop doing that if that makes you uncomfortable as removing `final` is indeed less safe that having it there, or adding it everywhere missing, even if someone just copy-pasted that there from a previous such existing activity. 🤷